viewing_party/party.py

```diff
@@ -1,23 +1,156 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
+
+    if title == None or genre == None or rating == None:
+        return None
+
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    movies = user_data["watchlist"]
+    for movie in range(len(movies)):
+        if movies[movie]['title'] == title:
+            add_to_watched = movies[movie]
+            del movies[movie]
+            user_data["watched"].append(add_to_watched)
+    return user_data
+    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0 
+    watched_list = user_data['watched']
+
+    if len(watched_list) == 0:
+        return 0.0 
+    for movie in watched_list:
+        sum += movie['rating']
+    
+    average = sum/(len(watched_list))
+    return average
+
 
+def get_most_watched_genre(user_data):
+    genres_dict = {}
+    counter = 0 
 
+    watched_list = user_data['watched']
+    list_length = len(watched_list)
+    
+    if list_length == 0:
+        return None
+    for movie in watched_list:
+        genre = movie['genre']
+        if genre not in genres_dict:
+            genres_dict[genre] = 1
+        else: 
+            genres_dict[genre] += 1
+
+    for key, value in genres_dict.items():
+        if counter < value:
+            counter += value
+            most_frequent_genre = key

```

Comment:
While we do get the most_frequent_genre returned correctly, in the future you'll want to declare most_frequent_genre outside of this if statement and outside of the for loop. **Usually we initiate a variable outside a for loop,** then we do comparing inside the for loop and reset the value of the variable based on the comparison. Finally, we'll return the variable.